Mathlib/NumberTheory/DirichletCharacter/Basic.lean

```diff
@@ -114,4 +116,73 @@ lemma mem_conductorSet_dvd {x : ℕ} (hx : x ∈ conductorSet χ) : x ∣ n := h
 The Dirichlet character `χ` can then alternatively be reformulated on `ℤ/nℤ`. -/
 noncomputable def conductor : ℕ := sInf (conductorSet χ)
 
+lemma conductor_mem_conductorSet : conductor χ ∈ conductorSet χ :=
+  Nat.sInf_mem (Set.nonempty_of_mem (level_mem_conductorSet χ))
+
+lemma conductor_dvd_level : conductor χ ∣ n := (conductor_mem_conductorSet χ).dvd
+
+lemma FactorsThrough_conductor : FactorsThrough χ (conductor χ) := conductor_mem_conductorSet χ
+
+lemma conductor_one (hn : 0 < n) : conductor (1 : DirichletCharacter R n) = 1 := by
```

Comment:
We can put one half of a later lemma before this lemma, to be used in the proof:
```lean
lemma conductor_ne_zero (hn : n ≠ 0) : conductor χ ≠ 0 :=
  fun h ↦ hn <| Nat.eq_zero_of_zero_dvd <| h ▸ conductor_dvd_level _

lemma conductor_one (hn : n ≠ 0) : conductor (1 : DirichletCharacter R n) = 1 := by
  suffices : FactorsThrough (1 : DirichletCharacter R n) 1
  · have h : conductor (1 : DirichletCharacter R n) ≤ 1 :=
      Nat.sInf_le <| (mem_conductorSet_iff _).mpr this
    exact Nat.le_antisymm h (Nat.pos_of_ne_zero <| conductor_ne_zero _ hn)
  · exact (FactorsThrough_one_iff _).mpr rfl
```

Reply:
Also, note that I am using the formally weaker `n ≠ 0` instead of `0 < n` as an assumption.
One could also think of using the `NeZero` typeclass.

Reply:
done